### DISCUSSION TITLE
Indirect modification of overloaded property has no effect

### DISCUSSION THREAD

**Discussion by bhushanc-reconnectin** [laravel/framework#50572] [Q&A]
I am getting an error "Indirect modification of overloaded property has no effect" in laravel v10.48.2, but same code is working fine in v10.45.1. Any idea what is the issue? tried to check respective release logs, but can not find anything.

Urgent help with this issue will be highly appreciated.

**Comment by henzeb** [comment]
any example code on how we can reproduce this will be highly appreciated.

**Comment by bhushanc-reconnectin** [comment]
Please find below the code snippets 

In Activity.php model I have below 1-to-1 relationship

public function activity_pattern(){
         return $this->hasOne(ActivityPattern::class,'activity_id','id')->orderBy('activity_id','ASC');
    }

In Routine.php model I have below relationship

public function activities_with_sorting(){
        return $this->belongsToMany(Activity::class, 'routine_activities','routine_id','activity_id')->where('routine_activities.status',1)->where('routine_activities.is_deleted',0)->orderBy('sorting', 'ASC');
    }

in controller I have below code:

$routines = Routine::with('activities_with_sorting.activity_pattern')
            ->where('id',$routineId)
            ->orderby('id','asc')
            ->first();

foreach ($routines->activities_with_sorting as $activity){

$customPattern = RoutineActivity::where('routine_id','=',$routineId)
                    ->where('activity_id','=',$activity->id)
                    ->where('status','=',1)
                    ->where('is_deleted','=',0)
                    ->first();

$dayOne = ($customPattern->day_one !=null) ? $customPattern->day_one : $activity->activity_pattern->day_one;

$activity->activity_pattern->day_one    = $dayOne;

}

Further, I have tried to downgrade the laravel version from v10.48.2 to v10.45.1 but still get same issue. 
PHP version is "PHP 8.1.13 (cli) (built: Nov 26 2022 14:07:36) (NTS)" where we are getting this error.

Thank you


Thank you
